### PR TITLE
Tnayin 3 (iterators)

### PR DESCRIPTION
**Pull request by GevorgeX**

### CODE REVIEW

File: Demo/Iterator/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+project(iterator)
+
+
+add_executable(iterator CircularIntIterator.cpp )
+#add_executable(iterator FilterIntegerIterator.cpp )
+#add_executable(iterator ReverseArrayIterator.cpp )
```
Comment (MikayelSmbatian) [inline Demo/Iterator/CMakeLists.txt]: Uncomment these lines.

File: Demo/Iterator/CircularIntIterator.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include <iostream>
+#include <vector>
+#include <Iterator/CircularIntIterator.h>
+
+int main() {
+        std::vector<int> numbers = {10, 20, 30};
```
Comment (MikayelSmbatian) [inline Demo/Iterator/CircularIntIterator.cpp]: Fix the idetation every where.

File: Utils/Iterator/FilterIntegerIterator.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return FilterIntegerIterator(m_ptr, (m_end_ptr- m_ptr)>>sizeof (int), m_fun);
+    }
+    FilterIntegerIterator end()
+    {
+        return FilterIntegerIterator(m_end_ptr, (m_end_ptr- m_ptr)>>sizeof (int), m_fun);
+    }
+private:
+    pointer m_end_ptr;
+    pointer m_ptr;
+    bool (*m_fun)(int);
```
Comment (MikayelSmbatian) [inline Utils/Iterator/FilterIntegerIterator.h]: To have more readable code consider declaring pointer to function like this:

```
using UnraryIntPredicate = bool(*)(int);
...
pointer m_end_ptr;
pointer m_ptr;
UnraryIntPredicate m_fun;
//bool (*m_fun)(int);
```

**Review by MikayelSmbatian — CHANGES_REQUESTED** [review]

**Comment by GevorgeX** [comment]
Uxxeci

**Review by MikayelSmbatian — APPROVED** [review]

**Pull request merged**